src/compiler/checker.ts

```diff
@@ -9438,7 +9443,12 @@ namespace ts {
                     error(node, Diagnostics.Value_of_type_0_is_not_callable_Did_you_mean_to_include_new, typeToString(funcType));
                 }
                 else {
-                    error(node, Diagnostics.Cannot_invoke_an_expression_whose_type_lacks_a_call_signature);
+                    if (apparentType.flags & TypeFlags.Union && forEach((<UnionType>apparentType).types, typeHasCallSignatures)) {
+                        // TODO check Not all types in '{0}' have compatible call signatures.

```

Comment:
Just get rid of this portion entirely and use it in the separate PR. I'd rather not temporarily disable error messages until another PR comes in.